sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -21,21 +29,141 @@ namespace Azure { namespace Core { namespace Test {
    */
   class TestBase : public ::testing::Test {
 
+  private:
+    // Call this method to update client options with the required configuration to
+    // support Record & Playback.
+    // If Playback or Record is not set, no changes will be done to the clientOptions or credential.
+    // Call this before creating the SDK client
+    void PrepareClientOptions(
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        Azure::Core::_internal::ClientOptions& options)
+    {
+      // Set up client options depending on the test-mode
+      if (m_testContext.IsPlaybackMode())
+      {
+        // Playback mode uses:
+        //  - playback transport adapter to read and return payload from json files
+        //  - never-expiring test credential to never require a token
+        options.Transport.Transport = m_interceptor->GetPlaybackClient();
```

Comment:
`GetPlaybackClient()` returns a `Transport`.  Would make more sense to call it `GetPlaybackTransport()`

Reply:
sure.